Classes/TYPO3/Imagine/ImagineFactory.php

```diff
@@ -67,11 +69,38 @@ public function create($className = 'Imagine')
             case 6:
                 $object = new $className($arguments[0], $arguments[1], $arguments[2], $arguments[3], $arguments[4], $arguments[5]);
                 break;
-            default:
-                $class = new \ReflectionClass($className);
+			default:
+				$class = new \ReflectionClass($className);
                 $object = $class->newInstanceArgs($arguments);
-        }
+		}
+
+		return $object;
+	}
+
+	/**
+	 * Set driver specific settings.	 
+	 *
+	 * @return void
+	 */
+	protected function configureDriverSpecificSettings() {
+		if ($this->settings['driver'] === 'Imagick') {
+			$this->configureImagickSettings();
+		}
+	}
+
+	/**
+	 * Sets limits for the Imagick driver.
+	 *
+	 * @return void
+	 */
+	protected function configureImagickSettings() {
+		if (!isset($this->settings['driverSpecific']['Imagick'])) {
+			return;
+		}
 
-        return $object;
-    }
+		$limits = $this->settings['driverSpecific']['Imagick']['limits'] ?? [];
```

Comment:
null coalesce operator only available in PHP 7.0+, Imagine 2.0 requires Flow ~3.0 which is 5.5